Assets/Src/Scripts/SeedCalc/Calculator.cs

```diff
@@ -0,0 +1,233 @@
+
+    [BindableProperty]
+    public CalculatorState State {
+      get => _state;
+      set => MvvmSetter(ref _state, value);
+    }
+
+    [BindableProperty]
+    public DisplayContent DisplayContent {
+      get => _displayContent;
+      set => MvvmSetter(ref _displayContent, value);
+    }
+
+    private CalculatorState _state;
+    private const int _maxChars = 100;
+    private readonly StringBuilder _cachedInput = new StringBuilder();
+    private DisplayContent _displayContent = null;
+
+    public void OnInput(string input) {
+      // A simple state machine.
+      if (State.IsError()) {
+        switch (input) {
+          case CalculatorInput.AllClear:
+            ResetState();
+            AllClear();
+            break;
+          case CalculatorInput.Del:
+            ResetState();
+            Backspace();
+            Parse();
+            break;
+          case CalculatorInput.Equal:
+          default:
+            // Neither a printable input nor the Equal key is accepted if the calculator is in an
+            // error state.
+            break;
+        }
+      } else {
+        switch (input) {
+          case CalculatorInput.AllClear:
+            AllClear();
+            break;
+          case CalculatorInput.Del:
+            Backspace();
+            Parse();
+            break;
+          case CalculatorInput.Equal:
+            Execute();
+            break;
+          case CalculatorInput.Dot:
+            // Adds the leading zero before a single dot.
+            if (_cachedInput.Length <= 0) {
+              Append(CalculatorInput.N0);
+            }
+            Append(CalculatorInput.Dot);
+            Parse();
+            break;
+          default:
+            Append(input);
+            if (State.IsOk()) {
+              Parse();
+            }
+            break;
+        }
+      }
+    }
+
+    // Resets the calculator state to OK.
+    private void ResetState() {
+      State = CalculatorState.Okay;
+    }
+
+    // Clears the input buffer and the tokens for display.
+    private void AllClear() {
+      _cachedInput.Clear();
+      DisplayContent = null;
+    }
+
+    // Appends the input character to the buffer.
+    private void Append(string input) {
+      _cachedInput.Append(input);
+      if (_cachedInput.Length > _maxChars) {
+        State = CalculatorState.Overflow;
+      }
+    }
+
+    // Removes the last character from the buffer.
+    private void Backspace() {
+      if (_cachedInput.Length > 0) {
+        _cachedInput.Remove(_cachedInput.Length - 1, 1);
+      }
+    }
+
+    // Parses the input string to an expression.
+    private void Parse() {
+      string expression = _cachedInput.ToString();
+      if (string.IsNullOrEmpty(expression)) {
+        DisplayContent = null;
+        return;
+      }
+      var collection = new DiagnosticCollection();
+      var executor = new Executor();
+      executor.Parse(expression, _moduleName, SeedXLanguage.Python, collection);
```

Comment:
Which language do we plan to use for calculator app? SeedPython or SeedBlock? I supposed it shall be SeedBlock.